edx_exams/apps/core/tests/test_api.py

```diff
@@ -119,6 +121,95 @@ def test_get_exam_attempt_missing_time_limit(self):
         self.assertEqual(0, time_left)
 
 
+@ddt.ddt
+class TestCheckIfExamTimedOut(ExamsAPITestCase):
+    """
+    Tests for API utility function `check_if_exam_timed_out`
+    """
+    # setup:
+
+    def setUp(self):
+        super().setUp()
+
+        self.course_id = 'course-v1:edx+test+f19'
+        self.content_id = 'block-v1:edX+test+2023+type@sequential+block@1111111111'
+
+        self.exam = Exam.objects.create(
+            resource_id=str(uuid.uuid4()),
+            course_id=self.course_id,
+            provider=self.test_provider,
+            content_id=self.content_id,
+            exam_name='test_exam',
+            exam_type='proctored',
+            time_limit_mins=60,
+            due_date='3000-07-01 00:00:00',
+            hide_after_due=False,
+            is_active=True
+        )
+
+        self.one_hour_ago = timezone.now() - timedelta(hours=1)
+
+    def create_mock_attempt(self, status, start_time, allowed_time_limit_mins):
+        return ExamAttempt.objects.create(
+            user=self.user,
+            exam=self.exam,
+            attempt_number=1,
+            status=status,
+            start_time=start_time,
+            allowed_time_limit_mins=allowed_time_limit_mins
+        )
+
+    @ddt.data(
+        # in progress and timed out
```

Comment:
NIT: instead of leaving a comment prior to the args, it's a little easier to read if the comment is next to the tuple! I think this is typically how we've commented on ddt cases